Sources/Optimizely/OptimizelyError.swift

```diff
@@ -48,12 +48,16 @@ public enum OptimizelyError: Error {
     case conditionNoMatchingAudience(_ id: String)
     case conditionInvalidFormat(_ hint: String)
     case conditionCannotBeEvaluated(_ hint: String)
-    case evaluateAttributeInvalidType(_ hint: String)
-    case evaluateAttributeValueOutOfRange(_ hint: String)
+    case evaluateAttributeInvalidCondition(_ condition: String)
+    case evaluateAttributeInvalidType(_ condition: String, _ value: Any, _ key: String)
+    case evaluateAttributeValueOutOfRange(_ condition: String, _ key: String)
     case evaluateAttributeInvalidFormat(_ hint: String)
-    case userAttributeInvalidType(_ hint: String)
-    case userAttributeInvalidMatch(_ hint: String)
-    case userAttributeInvalidFormat(_ hint: String)
+    case userAttributeInvalidType(_ condition: String)
+    case userAttributeInvalidMatch(_ condition: String)
+    case userAttributeNilValue(_ condition: String)
+    case nilAttributeValue(_ condition: String, _ key: String)
```

Comment:
Do we need both of these? Looks like redundant. If so, let's use "userAttributeNilValue" to be consistent.

Reply:
both serve different purposes: 
`userAttributeNilValue` = `Audience condition (\(condition)) evaluated to UNKNOWN because of null value.`
`nilAttributeValue` = `Audience condition (\(condition)) evaluated to UNKNOWN because a null value was passed for user attribute (\(key)).`